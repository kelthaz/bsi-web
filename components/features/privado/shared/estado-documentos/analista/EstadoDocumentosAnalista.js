import PropTypes from 'prop-types';
import SvgAprobado from '../../../../../svgs/estados/SvgAprobado';

import styles from '../estado-documentos.module.scss';

// ESTADOS: 0: Pendiente, 1: Revisado
const EstadoDocumentosAnalista = ({ prospecto }) => {

  const iconoEstado = (estado) => {
    switch (estado) {
      case 0:
        return <button className="btn-mini-secondary" type="button">Revisar</button>;
      case 1:
        return <SvgAprobado />;
      default:
        return <></>;
    }
  };

  return (
    <div>
      <div className="col-12">
        {prospecto.documentos && prospecto.documentos.map((documento) => (
          <div className="row my-3" key={documento.id}>
            <div className="col-5 my-auto"><span className="body2">{documento.nombre}</span></div>
            <div className={`col-4 my-auto ${styles.overme}`}>
              <span className="link" title={documento.ruta}>{documento.ruta}</span>
            </div>
            <div className="col-3 my-auto text-center">{iconoEstado(documento.estado)}</div>
          </div>
        ))}
        {prospecto.relaciones && prospecto.relaciones.map((relacion) => (
          <div key={relacion.id}>
            <div className="row my-3 card-simple-blue-light">
              <div className="col-12">
                <span className="d-flex sub color-blue-storm">{ relacion.nombre }</span>
                <span className="d-flex overline color-gray">{ relacion.descripcion }</span>
              </div>
            </div>
            {relacion.documentos.map((documento) => (
              <div className="row my-3" key={documento.id}>
                <div className="col-5 my-auto"><span className="body2">{documento.nombre}</span></div>
                <div className={`col-4 my-auto ${styles.overme}`}>
                  <span className="link" title={documento.ruta}>{documento.ruta}</span>
                </div>
                <div className="col-3 my-auto text-center">{iconoEstado(documento.estado)}</div>
              </div>
            ))}
          </div>
        ))}
      </div>
    </div>
  );
};

EstadoDocumentosAnalista.propTypes = {
  prospecto: PropTypes.object.isRequired
};

export default EstadoDocumentosAnalista;
