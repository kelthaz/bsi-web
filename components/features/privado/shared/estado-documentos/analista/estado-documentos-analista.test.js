import React from 'react';
import { shallow } from 'enzyme';
import EstadoDocumentosAnalista from './EstadoDocumentosAnalista';

describe('Pruebas en el componente EstadoDocumentosAnalista', () => {
  it('Debe mostrar el componente', () => {
    // arrange
    const prospecto = {
      documentos: [
        { nombre: 'Identificación oficial (Frente)', ruta: 'ALE_INE.jpg', estado: 1 },
        { nombre: 'Identificación oficial (Reverso)', ruta: 'ALE_INE.jpg', estado: 1 },
        { nombre: 'Comprobante de domicilio (Comercial)', ruta: 'comprobante_luz_factura_hogar.pdf', estado: 0 },
        { nombre: 'Comprobante de domicilio (Fiscal)', ruta: 'comprobante_luz_factura_hogar.pdf', estado: 0 },
        { nombre: 'Acta de matrimonio', ruta: 'Acta_matrimonio.pdf', estado: 0 },
        { nombre: 'Prueba de vida', ruta: 'prueba.mv', estado: 0 },
        { nombre: 'Selfie', ruta: 'selfie.png', estado: 0 },
        { nombre: 'Autorización de crédito', ruta: 'Autorizacion_credito.pdf', estado: 0 },
        { nombre: 'Reporte de buró de crédito', ruta: 'reporte_buro_credito.pdf', estado: 0 }
      ],
      relaciones: [
        {
          nombre: 'Carlos Pérez Díaz', descripcion: 'Cónyuge del solicitante',
          documentos: [
            { nombre: 'Identificación oficial (Frente)', ruta: 'INE_Carlos.jpg', estado: 1 },
            { nombre: 'Identificación oficial (Reverso)', ruta: 'INE_Carlos.jpg', estado: 1 }
          ]
        }
      ]
    };
    const wrap = shallow(<EstadoDocumentosAnalista prospecto={prospecto} />);
    // act
    // assert
    expect(wrap.find('span.body2').first().text()).toEqual(documentos[0].nombre);
    expect(wrap.find('.link').first().text()).toEqual(documentos[0].ruta);
  });
});
