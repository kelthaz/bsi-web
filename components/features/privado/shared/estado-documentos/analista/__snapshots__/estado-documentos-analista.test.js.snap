// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EstadoDocumentosAnalista Debe coincidir con la snapshot 1`] = `
<div>
  <div
    className="col-12"
  >
    <div
      key="1"
    >
      <div
        className="row my-3"
        key="10"
      >
        <div
          className="col-5 my-auto"
        >
          <span
            className="body2"
          >
            Identificación oficial (Frente)
          </span>
        </div>
        <div
          className="col-4 my-auto text-overflow"
        >
          <span
            className="link"
            title="INE_Carlos.jpg"
          >
            INE_Carlos.jpg
          </span>
        </div>
        <div
          className="col-3 my-auto text-center"
        >
          <SvgAprobado />
        </div>
      </div>
      <div
        className="row my-3"
        key="11"
      >
        <div
          className="col-5 my-auto"
        >
          <span
            className="body2"
          >
            Identificación oficial (Reverso)
          </span>
        </div>
        <div
          className="col-4 my-auto text-overflow"
        >
          <span
            className="link"
            title="INE_Carlos.jpg"
          >
            INE_Carlos.jpg
          </span>
        </div>
        <div
          className="col-3 my-auto text-center"
        >
          <SvgAprobado />
        </div>
      </div>
    </div>
  </div>
</div>
`;
